internal_packages/keybase/lib/private-key-popover.cjsx

```diff
@@ -0,0 +1,138 @@
+{React, Actions, AccountStore} = require 'nylas-exports'
+{remote} = require 'electron'
+Identity = require './identity'
+PGPKeyStore = require './pgp-key-store'
+PassphrasePopover = require './passphrase-popover'
+_ = require 'underscore'
+fs = require 'fs'
+pgp = require 'kbpgp'
+
+module.exports =
+class PrivateKeyPopover extends React.Component
+  constructor: ->
+    @state = {
+      selectedAddress: "0"
+      keyBody: ""
+      paste: false
+      import: false
+      validKeyBody: false
+    }
+
+  @propTypes:
+    addresses: React.PropTypes.array
+
+  render: =>
+    errorBar = <div className="invalid-key-body">Invalid key body.</div>
+    keyArea = <textarea value={@state.keyBody || ""} onChange={@_onKeyChange} placeholder="Paste in your PGP key here!"/>
+
+    saveBtnClass = if !(@state.validKeyBody) then "btn modal-done-button btn-disabled" else "btn modal-done-button"
+    saveButton = <button className={saveBtnClass} disabled={!(@state.validKeyBody)} onClick={@_onDone}>Save</button>
+
+    <div className="private-key-popover" tabIndex=0>
+      <span key="title" className="picker-title"><b>No PGP private key found.<br/>Add a key for {@_renderAddresses()}</b></span>
+      <div className="key-add-buttons">
+        <button className="btn btn-toolbar paste-btn" onClick={@_onClickPaste}>Paste in a Key</button>
+        <button className="btn btn-toolbar import-btn" onClick={@_onClickImport}>Import from File</button>
+      </div>
+      {if (@state.import or @state.paste) and !@state.validKeyBody and @state.keyBody != "" then errorBar}
```

Comment:
This totally works, but it can get a little confusing to see what the final state will be with so many conditional elements. If this were to become more complicated, I think we'd want to consider:

1. Setting a CSS class on the container, and then using CSS display:none to hide elements which aren't relevant, rather than making all the rendering conditional.

2. Identifying commonality between different states (maybe `(@state.import or @state.paste)`) and creating two render methods. Even though this might cause some duplication of elements that are present in both cases, find replace is easy and following conditionals is hard - just having renderImport / renderNonimport might be easier to maintain.